Add Drone Mods as Exocortex Mods

rmorgens opened the pull request.

LebombJames commented:
Do me a favour and pull from Dev and recook please.

rmorgens commented:
Yeah, I was seeing a lot of changes that I didn't touch making me think my
dev branch has issues, and hence my working branch....And I'm still trying
to figure out Git and VSCode, so wouldn't be surprised if there is operator
error involved....
For the spaces before links and inline embellishments, I'm guessing that
should be done after laying it in?  I've just been using the WYSIWYG editor
so didn't see the code having spacing issues, for which I apologize, and
I'll work that clean up

- Rob

On Tue, Jan 31, 2023, 9:11 AM LebombJames ***@***.***> wrote:

> Do me a favour and pull from Dev and recook please.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/foundryvtt-starfinder/foundryvtt-starfinder/pull/943#issuecomment-1410425451>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A5RI2V3W2WEKN4NKEVAVJKTWVEMQLANCNFSM6AAAAAAUKNOMVI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>

LebombJames commented:
Don't worry, I only want you to pull from Dev because of some changes I've made that will clean up your work a bit. Cook will also make it look like you've updated loads of files but it really hasn't, that's just Viscose being bad.

As for the HTML, the embellishments should be cleaned up on a HTML level (remove needless spans, remove style blocks etc). You can use the WYSIWYG editor as long as you're mindful of what's going into your HTML (protip: Ctrl+Shift+V pastes without copying needless HTML from the source)

rmorgens commented:
No, when I tried to spin off a new branch from my local dev to work the
toolkit submission, the cook tripped an error with a missing file that was
related to my mods, so something got where it isn't supposed to be, so I
need to get a clean dev trunk from your latest...


- Rob

Rob

On Tue, Jan 31, 2023, 10:19 AM LebombJames ***@***.***> wrote:

> Don't worry, I only want you to pull from Dev because of some changes I've
> made that will clean up your work a bit. Cook will also make it look like
> you've updated loads of files but it really hasn't, that's just Viscose
> being bad.
>
> As for the HTML, the embellishments should be cleaned up on a HTML level
> (remove needless spans, remove style blocks etc). You can use the WYSIWYG
> editor as long as you're mindful of what's going into your HTML (protip:
> Ctrl+Shift+V pastes without copying needless HTML from the source)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/foundryvtt-starfinder/foundryvtt-starfinder/pull/943#issuecomment-1410571069>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A5RI2V7XHBSTCCS4LYJMETTWVEUQLANCNFSM6AAAAAAUKNOMVI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>

LebombJames commented:
Sounds like you just need an `npm ci`

rmorgens commented:
From which branch, dev?  And how do I get those changes into my working
branch? Or do I just need to start from a clean branch and reload the items
into the compendium and fix all the links?

- Rob

On Tue, Jan 31, 2023, 10:26 AM LebombJames ***@***.***> wrote:

> Sounds like you just need an npm ci
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/foundryvtt-starfinder/foundryvtt-starfinder/pull/943#issuecomment-1410582243>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A5RI2VZCFWA5S6LXATWAOB3WVEVK7ANCNFSM6AAAAAAUKNOMVI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>

LebombJames commented:
Let's take this to Discord

LebombJames commented on src/items/class-features/exocortex_mod_speed_(ex).json:
```diff
@@ -0,0 +1,136 @@
+      "gmnotes": "",
+      "short": "",
+      "unidentified": "",
+      "value": "<p>&gt;@UUID[Compendium.sfrpg.class-features.z8iOan4RJXYvVhit]{Exocortex Mods (Ex)}</p>\n<p>Your exocortex increases your <span>land speed by 10 feet.</span></p><p style=\"box-sizing:border-box;user-select:text;margin:0.5em 0px;color:rgb(25, 24, 19);font-family:'Exo 2', sans-serif;font-size:13px;font-style:normal;font-variant-ligatures:normal;font-variant-caps:normal;font-weight:4 ...
```
This note isn't needed, there is already a modifier that adds to land speed.

LebombJames commented on src/items/class-features/exocortex_mod_weapon_proficiency_(ex).json:
```diff
@@ -0,0 +1,120 @@
+      "gmnotes": "",
+      "short": "",
+      "unidentified": "",
+      "value": "<p>&gt;@UUID[Compendium.sfrpg.class-features.z8iOan4RJXYvVhit]{Exocortex Mods (Ex)}</p>\n<p>Your exocortex <span>gains one of the following feats as a bonus feat (it must meet all the prerequisites of the chosen feat): Advanced Melee Weapon Proficiency, or Heavy Weapon Proficiency. You can select this mod multiple times; each time, you gains another of the available proficiency fe ...
```
Link to the feats inline in the sentence, rather than as links at the bottom.

LebombJames commented on src/items/class-features/greater_resistance_(ex_drone).json:
```diff
@@ -0,0 +1,102 @@
+      "gmnotes": "",
+      "short": "",
+      "unidentified": "",
+      "value": "<p>&gt;@UUID[Compendium.sfrpg.class-features.cQtmz297P5g6g5xv]{Advanced Mods}</p>\n<p>Choose one type of energy for which your drone already has the resistance mod. Increase your drone’s resistance to that energy type by 5. You can select this mod multiple times. Its effects do not stack. Each time you select this mod, it applies to a different energy type for which your drone has ...
```
Notes should always be an instruction, rather than a "need".
"Select the appropriate resistance in the Modifiers tab."

LebombJames commented on src/items/class-features/greater_resistance_(ex_drone).json:
```diff
@@ -0,0 +1,102 @@
+    "modifiers": [
+      {
+        "_id": "7ae69242-7e93-414c-b2aa-428535c7ede7",
+        "name": "New Modifier",
```
Empty modifier

LebombJames commented on src/items/class-features/invisibility_field_(ex_drone).json:
```diff
@@ -0,0 +1,86 @@
+      "gmnotes": "",
+      "short": "",
+      "unidentified": "",
+      "value": "<p>&gt;@UUID[Compendium.sfrpg.class-features.cQtmz297P5g6g5xv]{Advanced Mods}</p>\n<p>Once per day as a standard action, your drone can turn invisible for 10 minutes, as per the <em>invisibility</em> spell. If it makes an attack during this time, the invisibility ends. Your drone can spend 2 Resolve Points to use this ability again on the same day. Your drone must have the reactiv ...
```
Inline link

rmorgens commented:
I believe I've addressed all the comments.  Let me know if I missed something or created a new problem.

LebombJames commented on src/items/class-features/armor_slot_(ex_exocortex).json:
```diff
@@ -42,7 +42,7 @@
       "short": "",
       "unidentified": "",
-      "value": "<p>&gt;@UUID[Compendium.sfrpg.class-features.z8iOan4RJXYvVhit]{Exocortex Mods (Ex)}</p>\n<p>Your exocortex gains an armor upgrade slot for an armor upgrade (see page 204). If you do not already have an armor upgrade to install, you must purchase one separately. You can install the armor upgrade only if your exocortex has enough open upgrade slots to meet the upgrade’s requirements ...
+      "value": "<p>&gt; @UUID[Compendium.sfrpg.class-features.z8iOan4RJXYvVhit]{Exocortex Mods (Ex)}</p>\n<p>Your exocortex gains an armor upgrade slot for an armor upgrade (see page 204). If you do not already have an armor upgrade to install, you must purchase one separately. You can install the armor upgrade only if your exocortex has enough open upgrade slots to meet the upgrade’s requirement ...
```
Did you add these spaces after the `>`? No worries if not

rmorgens replied:
I did.  I thought you indicated to make there there was a space before the links.  I may have misunderstood.  I can fix if the > negates that need.

LebombJames replied:
The `>` is the exception to that unfortunately, there should be no space between the `>` and the link. Apologies.

LebombJames commented on src/items/class-features/invisibility_field_(ex_drone).json:
```diff
@@ -44,7 +44,7 @@
       "short": "",
       "unidentified": "",
-      "value": "<p>&gt;@UUID[Compendium.sfrpg.class-features.cQtmz297P5g6g5xv]{Advanced Mods}</p>\n<p>Once per day as a standard action, your drone can turn invisible for 10 minutes, as per the <em>invisibility</em> spell. If it makes an attack during this time, the invisibility ends. Your drone can spend 2 Resolve Points to use this ability again on the same day. Your drone must have the reactiv ...
+      "value": "<p>&gt;@UUID[Compendium.sfrpg.class-features.cQtmz297P5g6g5xv]{Advanced Mods}</p>\n<p>Once per day as a standard action, your drone can turn invisible for 10 minutes, as per the <em>invisibility</em> spell @UUID[Compendium.sfrpg.spells.YDXegEus8p0BnsH1]{Invisibility}. If it makes an attack during this time, the invisibility ends. Your drone can spend 2 Resolve Points to use this a ...
```
For the inline links, replace the original word with the link
e.g `as per the <em>@UUID[Compendium.sfrpg.spells.YDXegEus8p0BnsH1]{invisibility}</em> spell`

rmorgens replied:
Will fix the inline links.

If you could tell me how not to include the .db files, I'll try to comply.

LebombJames replied:
In VSCode, I usually just revert any changes to .db files after running `cook`, but alternatively you can stage everything you want to commit, which will leave out the `.db`s.

I think VSCode might even let you just "destage" the dbs and it'll automatically stage everything else.

rmorgens replied:
So, my current steps after finishing the changes and closing foundry, in VSCode:
npm run unpack
npm run cook
Then I stage all the changes using '+'
Commit the Changes
Sync the branch to Git

Do I need to selectively stage?

LebombJames replied:
That way works, just make sure you unstage all dbs before committing (you can unstage by folder, so you can just unstage the entire `packs` folder)

LebombJames commented on src/items/class-features/exocortex_mods_(ex).json:
```diff
@@ -37,7 +37,7 @@
       "short": "",
       "unidentified": "",
-      "value": "<p>&gt;@UUID[Compendium.sfrpg.class-features.cM54UpNXExgDSbq4]{Exocortex}</p>\n<p>Your exocortex allows you to apply any one of the following drone mods to yourself as if you were a drone with that mod installed: armor slot, cargo rack, climbing claws, enhanced senses, hydrojets, jump jets, resistance, smuggler’s compartment, speed, or weapon proficiency (gaining proficiency in ad ...
+      "value": "<p>&gt;@UUID[Compendium.sfrpg.class-features.cM54UpNXExgDSbq4]{Exocortex}</p>\n<p>Your exocortex allows you to apply any one of the following drone mods to yourself as if you were a drone with that mod installed: armor slot, cargo rack, climbing claws, enhanced senses, hydrojets, jump jets, resistance, smuggler’s compartment, speed, or weapon proficiency (gaining proficiency in ad ...
```
Replace the original text of the mod with the link, same as the invisibility links

LebombJames commented on src/items/class-features/weapon_proficiency_(ex_exocortex).json:
```diff
@@ -42,7 +42,7 @@
       "short": "",
       "unidentified": "",
-      "value": "<p>&gt;@UUID[Compendium.sfrpg.class-features.z8iOan4RJXYvVhit]{Exocortex Mods (Ex)}</p>\n<p>Your exocortex <span>gains one of the following feats as a bonus feat (it must meet all the prerequisites of the chosen feat): Advanced Melee Weapon Proficiency, or Heavy Weapon Proficiency. You can select this mod multiple times; each time, you gains another of the available proficiency fe ...
+      "value": "<p>&gt; @UUID[Compendium.sfrpg.class-features.z8iOan4RJXYvVhit]{Exocortex Mods (Ex)}</p>\n<p>Your exocortex gains one of the following feats as a bonus feat (it must meet all the prerequisites of the chosen feat): Advanced Melee Weapon Proficiency @UUID[Compendium.sfrpg.feats.54qpDSqRUsVRWOSZ]{Advanced Melee Weapon Proficiency (Combat)}, or Heavy Weapon Proficiency @UUID[Compendiu ...
```
Inline links again

LebombJames reviewed (changes requested):
There may have been some inline links I missed.
Also in future, please don't include `.db` files in your commits.

LebombJames commented:
Nice one

rmorgens commented:
Attempted resolution of the inline link vs text cleanup.  Removed the space after the '>' I added due to misunderstanding directives on space before links.
Please advise how to not include .db files per your request.

LebombJames commented:
See https://github.com/foundryvtt-starfinder/foundryvtt-starfinder/pull/943#discussion_r1103827387
